File: javascript/05_security_misconfiguration.js
// A05 - Security Misconfiguration. VULN: exposes /debug and disables helmet.
// VULN: A05 - verbose errors, open admin route
function createApp(express) {
  const app = express();
  app.disable("x-powered-by"); // only cosmetic
  app.get("/debug/env", (_, res) => res.json(process.env)); // VULN
  app.use((err, _req, res, _next) => res.status(500).send(err.stack)); // VULN
  return app;
}

module.exports = { createApp };

File: python/05_security_misconfiguration.py
"""A05 - Security Misconfiguration. VULN: debug mode + exposed stack traces."""
import os

# VULN: A05 - debug enabled in production
DEBUG = os.getenv("DEBUG", "true").lower() == "true"

def handle_error(exc):
    if DEBUG:
        return {"error": str(exc), "traceback": __import__("traceback").format_exc()}
    return {"error": "internal"}
